### PR TITLE
update WSClient plus fix IO timeout error

### PR DESCRIPTION
**Pull request by melekes**
DEPENDS ON https://github.com/tendermint/tendermint/pull/604

### CODE REVIEW

File: tm-bench/transacter.go [hunk 1]
```diff
@@ -124,13 +125,19 @@ func (t *transacter) sendLoop(connIndex int) {
 			for i := 0; i < t.Rate; i++ {
 				// each transaction embeds connection index and tx number
 				tx := generateTx(connIndex, txNumber)
+				paramsJson, err := json.Marshal(map[string]interface{}{"tx": hex.EncodeToString(tx)})
```
Comment (melekes) [inline tm-bench/transacter.go]: should we use tendermint#WSClient here (with reconnects/pingpongs) ? like if there is a reconnect, will the numbers be correct? i.e. we claim we sent 10000 txs, but due to reconnects only 5000 made it to the server

**Comment by zramsay** [comment]
tendermint/tendermint#604 has been merged to master & included in latest release. is this PR good to go?

File: tm-bench/glide.yaml [hunk 2]
```diff
@@ -6,17 +6,15 @@ import:
 - package: github.com/gorilla/websocket
 - package: github.com/pkg/errors
 - package: github.com/rcrowley/go-metrics
-- package: github.com/tendermint/go-rpc
-  version: develop
-  subpackages:
-  - types
 - package: github.com/tendermint/tendermint
+  version: 2fd8496bc109d010c6c2e415604131b500550e37
```
Comment (melekes) [inline tm-bench/glide.yaml]: set to `v0.11.0`

File: tm-bench/main.go [hunk 3]
```diff
@@ -138,10 +142,18 @@ func startNodes(endpoints []string, blockCh chan<- tmtypes.Header, blockLatencyC
 }
 
 func startTransacters(endpoints []string, connections int, txsRate int) []*transacter {
+
+	var hostHash [16]byte
+	if hostName , err := os.Hostname(); err != nil {
+		hostHash =	md5.Sum([]byte("127.0.0.1"))
```
Comment (melekes) [inline tm-bench/main.go]: formatting!

File: tm-bench/main.go [hunk 4]
```diff
@@ -138,10 +142,18 @@ func startNodes(endpoints []string, blockCh chan<- tmtypes.Header, blockLatencyC
...
+	} else {
+		hostHash = md5.Sum([]byte(hostName))
```
Comment (melekes) [inline tm-bench/main.go]: code duplication!

File: tm-bench/main.go [hunk 5]
```diff
@@ -138,10 +142,18 @@ func startNodes(endpoints []string, blockCh chan<- tmtypes.Header, blockLatencyC
...
+	}
+
 	transacters := make([]*transacter, len(endpoints))
 
 	for i, e := range endpoints {
-		t := newTransacter(e, connections, txsRate)
+		t := newTransacter(e, connections, txsRate, hostHash)
```
Comment (melekes) [inline tm-bench/main.go]: don't like this. we can probably just call os.GetHostName inside transacter

File: tm-bench/transacter.go [hunk 6]
```diff
@@ -104,6 +108,17 @@ func (t *transacter) receiveLoop(connIndex int) {
 // sendLoop generates transactions at a given rate.
 func (t *transacter) sendLoop(connIndex int) {
 	c := t.conns[connIndex]
+
+	c.SetPingHandler(func(message string) error {
```
Comment (melekes) [inline tm-bench/transacter.go]: Why do we need to overwrite default?

Comment (melekes) [reply tm-bench/transacter.go]: oh, because of the sendTimeout...

File: tm-monitor/glide.lock [hunk 7]
```diff
@@ -45,7 +51,7 @@ imports:
   subpackages:
   - data
 - name: github.com/tendermint/tendermint
-  version: e9b7221292afe25ce956ea85ab83bb5708eb2992
+  version: 2fd8496bc109d010c6c2e415604131b500550e37
```
Comment (melekes) [inline tm-monitor/glide.lock]: v0.11.0

**Pull request merged**